### PR TITLE
Clean up push proposer setting method

### LINKED ISSUE #11089: Clean up validator push proposer and registration settings
Let's clean up the following methods to be more Idiomatic.
- `PushProposerSettings`
- `buildProposerSettingsRequests`
- `cacheValidatorPubkeyHexToValidatorIndex`

### PR DESCRIPTION
**Pull request by terencechain**
Fixes #11089

This PR splits `buildProposerSettingsRequests` into `buildPrepProposerReqs` and `buildSignedRegReqs`. One for the general prepare proposer route and one for the special validator registration MEV route. The separation is nice and keeps the code clean with future additions (i.e. relayer endpoints). More details will be commented through out the PR

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/server.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (vs *Server) ValidatorIndex(ctx context.Context, req *ethpb.ValidatorIndexR
 	}
 	index, ok := st.ValidatorIndexByPubkey(bytesutil.ToBytes48(req.PublicKey))
 	if !ok {
-		return nil, status.Errorf(codes.Internal, "Could not find validator index for public key %#x not found", req.PublicKey)
+		return nil, status.Errorf(codes.NotFound, "Could not find validator index for public key %#x not found", req.PublicKey)
```
Comment (terencechain) [inline beacon-chain/rpc/prysm/v1alpha1/validator/server.go:127]: `NotFound` is more appropriate here and enables better error handling for the caller

File: validator/client/registration.go [hunk 2]
```diff
@@ -74,7 +73,6 @@ func (v *validator) SignValidatorRegistrationRequest(ctx context.Context, signer
 	} else {
 		sig, err := signValidatorRegistration(ctx, signer, newValidatorRegistration)
 		if err != nil {
-			log.WithError(err).Error("failed to sign builder validator registration obj")
```
Comment (terencechain) [inline validator/client/registration.go:77]: More of a caller's responsibility to log this. Logging it on the caller side now

File: validator/client/validator.go [hunk 3]
```diff
@@ -950,150 +952,169 @@ func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKey
...
 	resp, err := v.validatorClient.ValidatorIndex(ctx, &ethpb.ValidatorIndexRequest{PublicKey: pubkey[:]})
-	if err != nil {
-		hexKey := hexutil.Encode(pubkey[:])
-		if strings.Contains(err.Error(), "Could not find validator index") {
-			log.Warnf("Could not find validator index for public key %#x not found. "+
-				"Perhaps the validator is not yet active.", hexKey)
-			return 0, false, nil
-		}
+	switch {
+	case status.Code(err) == codes.NotFound:
```
Comment (terencechain) [inline validator/client/validator.go:1113]: New error handling

File: validator/client/validator.go [hunk 4]
```diff
@@ -950,150 +952,169 @@ func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKey
...
-		log.WithFields(logrus.Fields{
-			"totalNonActivePubkeys": len(pubkeys) - len(proposerFeeRecipientRequests),
-		}).Warnln("will not prepare beacon proposer and update fee recipient until a validator index is assigned")
-	}
-
-	if len(signedRegisterValidatorRequests) != len(pubkeys) && anyValidatorRegistrationEnabled {
-		log.WithFields(logrus.Fields{
-			"totalNonActivePubkeys": len(pubkeys) - len(signedRegisterValidatorRequests),
-		}).Warnln("will not be included in builder validator registration until a validator index is assigned")
-	}
```
Comment (terencechain) [inline validator/client/validator.go:1083]: I completely removed this warning log. It feels it'd be prone to miss information. It's normal for `len(pubkeys) != len(req)` given any validator could be `enabled: false`

File: validator/client/validator_test.go [hunk 5]
```diff
@@ -1984,78 +1983,6 @@ func TestValidator_PushProposerSettings(t *testing.T) {
 			},
 			err: "could not submit signed registrations to beacon node",
 		},
-		{
-			name: "Validator Index Not found with validator registration",
```
Comment (terencechain) [inline validator/client/validator_test.go:1988]: This test is outdated. The validator index is irrelevant for validator registration. Validator registration doesn't use the validator index. It only uses public key

**Review by james-prysm — APPROVED** [review]
Originally I had it all in one function to save the looking of the public keys but things did get a lot more complex, it's probably not that expensive to go over one more time and does make things easier to read. Thanks for the cleanup.

**Comment by terencechain** [comment]
Let's wait until after the Prater RC

**Pull request merged**